pylib/System/TestDef.py

```diff
@@ -437,21 +437,20 @@ def configTest(self):
             if not os.path.isfile(testFile):
                 print("Test .ini file not found!: " + testFile)
                 sys.exit(1)
-##TO DO, remove from loop, change testFile to list of files               
-            self.config = configparser.ConfigParser()
-            self.config.read(testFile)
-            for section in self.config.sections():
-                if self.logger is not None:
-                    self.logger.verbose_print("SECTION: " + section)
-                    self.logger.verbose_print(self.config.items(section))
-                if self.options['dryrun']:
-                    continue
-                if section.startswith("SKIP") or section.startswith("skip"):
-                    # users often want to temporarily ignore a section
-                    # of their test definition file, but don't want to
-                    # remove it lest they forget what it did. So let
-                    # them just mark the section as "skip" to be ignored
-                    continue;
+        self.config = configparser.ConfigParser()
+        self.config.read(self.log['inifiles'])
```

Comment:
I put a comment in the TestDef.py change file,  I am not sure if you get an
email when that happens or not.  Here are the contents of the comment:

self.config.read(testFile) was part of the loop checking to see if the .ini
files existed. I took the self.config.read out of the loop and used the
list of files from the new argparse args.ini_files which was producing a
list of lists so I needed to add the [0] to get to the actual list of
files. ConfigParser will take either a single file or a list of files but
it will not take a list of lists.

Deb Rezanka

On Tue, Jul 12, 2016 at 11:52 AM, rhc54 <notifications@github.com> wrote:

> In pylib/System/TestDef.py
> <https://github.com/open-mpi/mtt/pull/453#discussion_r70486982>:
>
> > -            self.config = configparser.ConfigParser()
> > -            self.config.read(testFile)
> > -            for section in self.config.sections():
> > -                if self.logger is not None:
> > -                    self.logger.verbose_print("SECTION: " + section)
> > -                    self.logger.verbose_print(self.config.items(section))
> > -                if self.options['dryrun']:
> > -                    continue
> > -                if section.startswith("SKIP") or section.startswith("skip"):
> > -                    # users often want to temporarily ignore a section
> > -                    # of their test definition file, but don't want to
> > -                    # remove it lest they forget what it did. So let
> > -                    # them just mark the section as "skip" to be ignored
> > -                    continue;
> > +        self.config = configparser.ConfigParser()
> > +        self.config.read(self.log['inifiles'])
>
> I'm a little confused as to how this solves the problem. In line 435, we
> set self.log['inifiles'] to only args.ini_files[0]. Is that value the
> first file in the list? Or is it the entire list? If the former, then how
> does this result in reading all the .ini files?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-mpi/mtt/pull/453/files/e0666f12f3706421b00e1b2579d67654ec2b8c30#r70486982>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHhLWQgni88Cs0_Rf5xUHo4MB6zQZVuPks5qU9RIgaJpZM4JKnB4>
> .
>